atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1348,13 +1351,27 @@ private void createTablesWithLock(final Map<TableReference, byte[]> tableNamesTo
 
                     TableReference tableRefLowerCased = TableReference.createUnsafe(table.getQualifiedName().toLowerCase());
                     if (!existingTablesLowerCased.contains(tableRefLowerCased)) {
-                        client.system_add_column_family(getCfForTable(table, tableNamesToTableMetadata.get(table)));
+                        CfDef cfToCreate = getCfForTable(table, tableNamesToTableMetadata.get(table));
+
+                        int consistentId = ThreadLocalRandom.current().nextInt(1, Integer.MAX_VALUE);
+                        cfToCreate.setId(consistentId);
+                        cfToOurCfId.put(cfToCreate, consistentId);
+
+                        client.system_add_column_family(cfToCreate);
                     } else {
                         log.warn("Ignored call to create a table ({}) that already existed (case insensitive).", table);
                     }
                 }
                 if (!tablesToCreate.isEmpty()) {
                     CassandraKeyValueServices.waitForSchemaVersions(client, "(all tables in a call to createTables)", configManager.getConfig().schemaMutationTimeoutMillis());
+                    for (Entry<CfDef, Integer> cfAndCfId : cfToOurCfId.entrySet()) {
+                        String cfName = cfAndCfId.getKey().getName();
+                        int consistentIdAroundCluster = CassandraKeyValueServices.checkCfIdOnAllHosts(clientPool, keyspace, cfName);
+                        if (consistentIdAroundCluster != cfAndCfId.getValue()) {
+                            log.error("Tried to create a table " + cfName + "; we expected its CfId to be " + cfAndCfId.getValue() +
+                                    " but it looks like someone else created it successfully and fully consistently around the cluster before we did with CfId " + consistentIdAroundCluster);
+                        }
+                    }
```

Comment:
Should we do similar consistency check anywhere we `system_update_column_family` (`alterGcAndTombstone` and `internalPutMetadataForTables`)